bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java

```diff
@@ -845,6 +854,30 @@ protected void release(boolean wait)
       }
     }
 
+    private boolean canRelease(boolean writer)
```

Comment:
For a method named `canRelease()` I would expect the following sequence:
```
if (canRelease()) {
  release();
}
```
This method returns true if block can be evicted from memory.